ignite/metrics/gan/inception_score.py

```diff
@@ -0,0 +1,109 @@
+from ignite.metrics.gan.utils import InceptionModel
+
+# These decorators helps with distributed settings
+from ignite.metrics.metric import Metric, reinit__is_reduced, sync_all_reduce
+
+__all__ = ["InceptionScore"]
+
+
+class InceptionScore(Metric):
+    r"""Calculates Inception Score.
+
+    .. math::
+       \text{IS(G)} = \frac{1}{N}\sum_{i=1}^{N} D_{KL} (p(y|x^{(i)} \parallel \hat{p}(y)))
+
+    where :math:`p(y|x)` is the conditional probability of image being the given object and
+    :math:`p(y)` is the marginal probability that the given image is real.
+
+    More details can be found in `Barratt et al. 2018`__.
+
+    __ https://arxiv.org/pdf/1801.01973.pdf
+
+
+    Args:
+        num_probabilities: number of probabilities predicted by the model or number of classes of the model
+        output_transform: a callable that is used to transform the
+            :class:`~ignite.engine.engine.Engine`'s ``process_function``'s output into the
+            form expected by the metric. This can be useful if, for example, you have a multi-output model and
+            you want to compute the metric with respect to one of the outputs.
+            By default, metrics require the output as ``(y_pred, y)`` or ``{'y_pred': y_pred, 'y': y}``.
+        device: specifies which device updates are accumulated on. Setting the
+            metric's device to be the same as your ``update`` arguments ensures the ``update`` method is
+            non-blocking. By default, CPU.
+
+    Example:
+
+        .. code-block:: python
+
+            from ignite.metric.gan.IS import InceptionScore
+            import torch
+
+            probabilities = torch.rand(10, 2048), torch.rand(10, 2048)
+
+            m = InceptionScore(num_probabilities=1000)
+            m.update(probabilities)
+            print(m.compute())
+
+    .. versionadded:: 0.5.0
+    """
+
+    def __init__(
+        self,
+        num_probabilities: Optional[int] = None,
+        prediction_model: Optional[Callable] = None,
+        output_transform: Callable = lambda x: x,
+        device: Union[str, torch.device] = "cpu",
+    ) -> None:
+
+        if num_probabilities is None and prediction_model is None:
+            num_probabilities = 1000
+            prediction_model = InceptionModel(return_features=False)
+        elif num_probabilities is None:
+            raise ValueError("Argument num_probabilities should be defined")
+        elif prediction_model is None:
+            self._prediction_model = lambda x: x
+            prediction_model = self._prediction_model
+
+        if num_probabilities <= 0:
+            raise ValueError(f"Argument num_probabilities must be greater to zero, got: {num_probabilities}")
+        self._num_probs = num_probabilities
+        self._eps = 1e-16
+        super(InceptionScore, self).__init__(output_transform=output_transform, device=device)
+
+    @staticmethod
+    def _check_feature_input(samples: torch.Tensor) -> None:
```

Comment:
Add a test wrt `self._num_probs ?`

Reply:
Well, we are declaring the tensors storing sum of values based on `num_probs` in IS and `num_features` in FID so it will throw an error for wrong size, but we don't have explicit tests. I will write those for both.